UndertaleModTool/Scripts/Community Scripts/TouchControlsEnabler.csx

```diff
@@ -0,0 +1,411 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Linq;
+using UndertaleModLib.Util;
+
+EnsureDataLoaded();
+
+
+if (Data?.GeneralInfo?.DisplayName?.Content.ToLower() == "deltarune chapter 1 & 2" || Data?.GeneralInfo?.DisplayName?.Content.ToLower() == "deltarune chapter 1&2" || Data?.GeneralInfo?.DisplayName?.Content.ToLower() == "undertale")
+{
+    // nothing
+} else {
+    ScriptError("Error 0: Only compatible with Undertale & Deltarune");
+    return;
+}
+
+string dataPath = Path.Combine(Path.GetDirectoryName(ScriptPath), "TouchControls_data");
+
+int lastTextPage = Data.EmbeddedTextures.Count - 1;
+int lastTextPageItem = Data.TexturePageItems.Count - 1;
+
+Dictionary<string, UndertaleEmbeddedTexture> textures = new Dictionary<string, UndertaleEmbeddedTexture>();
+
+UndertaleEmbeddedTexture controls = new UndertaleEmbeddedTexture();
+controls.Name = new UndertaleString("Texture " + ++lastTextPage);
+controls.TextureData.TextureBlob = File.ReadAllBytes(dataPath+"/controls.png");
+Data.EmbeddedTextures.Add(controls);
+textures.Add(Path.GetFileName(dataPath+"/controls.png"), controls);
+
+UndertaleEmbeddedTexture black = new UndertaleEmbeddedTexture();
+black.Name = new UndertaleString("Texture " + ++lastTextPage);
+black.TextureData.TextureBlob = File.ReadAllBytes(dataPath+"/black.png");
+Data.EmbeddedTextures.Add(black);
+textures.Add(Path.GetFileName(dataPath+"/black.png"), black);
+
+UndertaleTexturePageItem pg_joybase1 = new UndertaleTexturePageItem();
+pg_joybase1.Name = new UndertaleString("PageItem " + ++lastTextPageItem);
+pg_joybase1.SourceX = 0; pg_joybase1.SourceY = 0; pg_joybase1.SourceWidth = 59; pg_joybase1.SourceHeight = 59;
+pg_joybase1.TargetX = 0; pg_joybase1.TargetY = 0; pg_joybase1.TargetWidth = 59; pg_joybase1.TargetHeight = 59;
+pg_joybase1.BoundingWidth = 59; pg_joybase1.BoundingHeight = 59;
+pg_joybase1.TexturePage = textures["controls.png"];
```

Comment:
Why are you using this syntax, when you use the other one for sprites later (see line 197) ?
You should make the syntax consistent.

Reply:
I'm not sure I understand the issue, kindly re-phrase it so that I can understand

Reply:
I mean, why you can't do that like this:
```csharp
UndertaleTexturePageItem pg_joybase1 = new UndertaleTexturePageItem()
{
  Name = new UndertaleString("PageItem " + ++lastTextPageItem),
  SourceX = 0, SourceY = 0, SourceWidth = 59, SourceHeight = 59,
  TargetX = 0, TargetY = 0, TargetWidth = 59, TargetHeight = 59,
  BoundingWidth = 59, BoundingHeight = 59,
  TexturePage = textures["controls.png"]
}
```

Reply:
The same as you did below that chunk of code for `UndertaleSprite`s

Reply:
ah good idea, I'll update it soon

Reply:
@VladiStep updated the syntax, sorry for the delay I got busy